### DISCUSSION TITLE
Best Example for Quick Start

### DISCUSSION THREAD

**Discussion by joelebgm** [riverqueue/river#424] [General]
Is there a very basic  example program that can be used to ensure that dependencies, database connections and a very simple job are working? I do see the example programs but am not sure which one is the best to start with. Also the examples I tried resulted in the error message "use of internal package ... not allowed". Many thanks in advance!

**Comment by brandur** [comment]
Hey @joelebgm, did you see the docs "getting started" page? It probably has the most comprehensive set of instructions needed to get up and running:

https://riverqueue.com/docs

The insert and work example [1] should be fairly easily adaptable to running code. Unfortunately we had to include a few internal constructs in there because the examples have to be runnable by Go's test suite, which is a little tricky when they need to connect to a real database. The sites you'll change will mostly involve removing `riverinternaltest` statements and replacing them with things like a real URL to your local Postgres. Repurposing one of these examples while crossreferencing the docs above should get you a long way.

[1] https://github.com/riverqueue/river/blob/master/example_insert_and_work_test.go

**Comment by joelebgm** [comment]
I did get it working although I had to clone the River repo as opposed to
creating a standalone program. Thanks!
